segmentation/utils/spatial_analysis.py

```diff
@@ -63,47 +68,129 @@ def calculate_channel_spatial_enrichment(dist_matrix, marker_thresholds, all_pat
                              "perimeter", "fov"]
 
     # Error Checking
-    if not np.isin(excluded_colnames, all_patient_data.columns).all():
+    if not np.isin(excluded_colnames, all_data.columns).all():
         raise ValueError("Column names were not found in Expression Matrix")
 
-    if not np.isin(points, all_patient_data.iloc[:, patient_idx]).all():
+    if not np.isin(fovs, all_data[fov_col]).all():
         raise ValueError("Points were not found in Expression Matrix")
 
     # Subsets the expression matrix to only have marker columns
-    data_markers = all_patient_data.drop(excluded_colnames, axis=1)
+    data_markers = all_data.drop(excluded_colnames, axis=1)
     # List of all markers
     marker_titles = data_markers.columns
     # Length of marker list
     marker_num = len(marker_titles)
 
     # Create stats Xarray with the dimensions (points, stats variables, number of markers, number of markers)
-    stats_raw_data = np.zeros((num_points, 7, marker_num, marker_num))
-    coords = [points, ["z", "muhat", "sigmahat", "p_pos", "p_neg", "h", "p_adj"], marker_titles,
+    stats_raw_data = np.zeros((num_fovs, 7, marker_num, marker_num))
+    coords = [fovs, ["z", "muhat", "sigmahat", "p_pos", "p_neg", "h", "p_adj"], marker_titles,
               marker_titles]
     dims = ["points", "stats", "marker1", "marker2"]
     stats = xr.DataArray(stats_raw_data, coords=coords, dims=dims)
 
     # Subsetting threshold matrix to only include column with threshold values
     thresh_vec = marker_thresholds.iloc[:, 1]
 
-    for point in points:
+    for i in range(0, len(fovs)):
         # Subsetting expression matrix to only include patients with correct label
-        patient_ids = all_patient_data.iloc[:, patient_idx] == point
-        patient_data = all_patient_data[patient_ids]
+        patient_ids = all_data[fov_col] == fovs[i]
+        fov_data = all_data[patient_ids]
         # Patients with correct label, and only columns of markers
-        patient_data_markers = data_markers[patient_ids]
-        # Subsetting the column with the cell labels
-        label_idx = patient_data.iloc[:, cell_label_idx]
+        fov_channel_data = data_markers[patient_ids]
+
+        # Subset the distance matrix array to only include the distance matrix for the correct point
+        # dist_matrix = dist_matrices[i, :, :].values
+        dist_matrix = dist_matrices[dist_mat_list[i]].value
```

Comment:
We want to use the keys() function of the H5 file to make sure that the distance matrix we're subsetting matches the name fo the FOV we're currently analyzing.